src/datachain/lib/projects.py

```diff
@@ -0,0 +1,81 @@
+from typing import Optional
+
+from datachain.error import ProjectCreateNotAllowedError
+from datachain.lib.namespaces import get as get_namespace
+from datachain.project import Project
+from datachain.query import Session
+
+
+def create(
+    name: str,
+    namespace_name: str,
+    description: Optional[str] = None,
+    session: Optional[Session] = None,
+) -> Project:
+    """
+    Creates new custom project.
+    Note that creating projects is not allowed for local environment, unlike in
+    Studio where it is allowed.
+    In local environment all datasets are created under default `local` project.
+    Project is also connected to it's parent namespace.
+
+    Parameters:
+        name : project name.
+        namespace : namespace name under which we are creating new project.
+        description : project description.
+        session : Session to use for creating project.
+
+    Example:
+        ```py
+        import datachain as dc
+        project = dc.create_project("my-project", "dev", "My personal project")
+        ```
+    """
+    session = Session.get(session)
+
+    if not session.catalog.metastore.project_allowed_to_create:
```

Comment:
these checks should be handled by the create_project below (we are leaking too much for now reason, all these `session.catalog.metastore.project_allowed_to_create` look too complicated)

Reply:
Hmm ... if `session.catalog.metastore.create_project` allows dataset to be created, then it is a very easy bypass and someone has as dataset in a local database. Do we want to allow that? This logic (allow or not) should be deeper, even to enforce the behavior we need. Unless I'm missing something.